### PR TITLE
Add promotional banner for new Firefox for Android [fix #9302]

### LINKED ISSUE #9302: [8/27] Banner promo for Fenix release
Description: The new release, Fenix, for Firefox mobile on Android is currently planned for global availability on the 27th. Promoting that on www.mozilla.org can be achieved by using the banner component on the site and placing that on high trafficked pages.

[Principal Project Doc](https://docs.google.com/document/d/1cmqftlLoUphUD7eGblUyAiapksP6fpy-PFGU-fORAyM/edit)

#### Success Criteria
- [ ] Banner is live on 2020-8-27 on the homepage, /firefox/, and /firefox/mobile/

#### Tasks
- [ ] Copy - Rob
- [ ] Design - Cristina
- [ ] Code - Craig

#### Approvals
- [ ] Accountable - Mei
- [ ] TL - Hoosteeno
- [ ] Brand copy - Natalie
- [ ] Brand design - Clio
- [ ] NA Brand TL - Eddie

### PR DESCRIPTION
**Pull request by craigcook**
## Description
Adds a banner promoting Daylight on a few key pages, behind the switch `firefox-daylight-launch-banner`

This also adds the names "Firefox Daylight" and "Daylight" to brands.ftl

Daylight is already available in most of the world but will release to US and Canada on **27 August** so we want to get the banner live in prod on that day if at all possible. Sorry for the last-minuteness, this kinda fell through the cracks with all the happenings.

## Issue / Bugzilla link
#9302 

## Testing
http://localhost:8000/
http://localhost:8000/firefox/
http://localhost:8000/firefox/browsers/
http://localhost:8000/firefox/products/
http://localhost:8000/firefox/mobile/

### CODE REVIEW

**Comment by craigcook** [comment]
OK, I don't know why Circle is failing now. It's the exact same error as [my other PR](https://github.com/mozilla/bedrock/pull/9326) and I couldn't figure it out there either. 

It seems to be somehow related to the presence of `switch()` on `firefox/index-master.html` combined with the `follow_redirects=True` [in these two redirect tests](https://github.com/mozilla/bedrock/blob/master/tests/redirects/map_globalconf.py#L989-L990). If I remove switches from that page, the tests pass. Those are the only two lines with `follow_redirects=True` but if just remove that they still fail so maybe it's something else.

I'm stumped and think I need so summon @pmac for help.

**Review by alexgibson — CHANGES_REQUESTED** [review]
The banner itself looks great. The only wrinkle is the home page view, which is currently a bit of a mess because of the mixture of hard-coded templates, fluent files, and .lang templates. I'm a little worried that once it hits production then activation may get messed up.

We could re-work the view entirely, but I wonder if adding this banner as a [global .ftl file](https://github.com/mozilla/bedrock/blob/master/bedrock/settings/base.py#L248-L257) may simplify things, seeing as it appears on so many pages and templates?

Any thoughts here, @pmac?

**Comment by alexgibson** [comment]
I'm stumped by those two redirect tests also. I'm not sure why they are failing, but I also don't understand what they are actually testing either? They seem to be leftovers from the old mozilla.com site, so I'm not quite sure what purpose they are serving today.

File: bedrock/mozorg/views.py [hunk 1]
```diff
@@ -109,28 +108,26 @@ def namespaces(request, namespace):
...
-    # we need it as a list.
-    donate_params['preset_list'] = donate_params['presets'].split(',')
     ctx = {
-        'donate_params': donate_params,
         'pocket_articles': PocketArticle.objects.all()[:4]
     }
 
     if locale.startswith('en-'):
         template_name = 'mozorg/home/home-en.html'
+        ctx['ftl_files'] = ['mozorg/home/home-en', 'banners/firefox-daylight-launch']
```
[the remark this answers is not in the document]
Comment (craigcook) [reply bedrock/mozorg/views.py]: 🤦 I very briefly entertained the notion of migrating all the home pages to Fluent before it dawned on me how dumb that was for the EN/FR/DE ones, but still had the view all set up for it.

**Review by alexgibson — APPROVED** [review]
r+ this looks good to me, and the global file should help to simplify things (we'll just need to remember to remove it once the promo finishes).

Only thing remaining is the mystery test failure, which I still don't quite understand.

**Comment by pmac** [comment]
> We could re-work the view entirely, but I wonder if adding this banner as a [global .ftl file](https://github.com/mozilla/bedrock/blob/master/bedrock/settings/base.py#L248-L257) may simplify things, seeing as it appears on so many pages and templates?

That's fine with me. Either a specific .ftl file for this banner that you manually include in every view, or a global is fine. Global might be better if we expect it to expand beyond this initial set of pages.

**Pull request merged**